easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -318,6 +319,73 @@ def configure_step(self, srcdir=None, builddir=None):
 
         (out, _) = run_cmd(command, log_all=True, simple=False)
 
+    def sanitycheck_for_configuration(self, out):
+        self.log.info("Checking Python paths")
+
+        if LooseVersion(self.cmake_version) >= '3.16':
+            try:
+                with open('CMakeCache.txt', 'r') as file:
+                    lines = file.readlines()
+            except FileNotFoundError:
+                self.log.warning("CMakeCache.txt not found. Python paths checks skipped.")
+                return
+
+            if not os.getenv('EBROOTPYTHON'):
+                self.log.warning("EBROOTPYTHON is not set.")
+                return
+            else:
+                self.log.info("Skipping Python path checks as CMake version might be lower than 3.16.")
+                return
+
+        python_exec_path = []
+        python_include_path = []
+        python_library_path = []
+
+        python_prefixes = r"(Python|Python2|Python3)_"
+
+        for line in lines:
+            if line.startswith(python_prefixes + r"EXECUTABLE:FILEPATH="):
+                python_exec_path.append(line.split('=')[1].strip())
+                self.log.info("Python executable path: %s", python_exec_path[-1])
+            elif line.startswith(python_prefixes + r"INCLUDE_DIR:FILEPATH="):
+                python_include_path.append(line.split('=')[1].strip())
+                self.log.info("Python include path: %s", python_include_path[-1])
+            elif line.startswith(python_prefixes + r"LIBRARY:FILEPATH="):
+                python_library_path.append(line.split('=')[1].strip())
+                self.log.info("Python library path: %s", python_library_path[-1])
+
+            # Check if paths are found and validate paths for symlinks
+            for path_list in [python_exec_path, python_include_path, python_library_path]:
+                for path in path_list:
+                    if path and not os.path.exists(path):
+                        raise EasyBuildError(f"Path does not exist: {path}")
+
+            # Check if paths are found and handle EBROOTPYTHON cases
+            if any(path for path in [python_exec_path, python_include_path, python_library_path]):
+                if not os.getenv('EBROOTPYTHON'):
+                    self.log.error(
+                        "EBROOTPYTHON is not set and Python related paths are found."
+                    )
+                elif not all(
+                    (path and os.getenv('EBROOTPYTHON') in path)
+                    for path in [python_exec_path, python_include_path, python_library_path]
```

Comment:
You are checking only for `$EBROOTPYTHON` being in a list of paths, which likely is always false. Maybe rename the variables to be able to spot that: `python_exec_path` -> `python_exec_paths` and so on.

It might even make sense to show which path is "wrong". I.e. which of the 3 lists and which exact path is outside EB.